### ISSUE TITLE
aws_imagebuilder_image_recipe creates components in random order

### ISSUE THREAD

**Issue by mpoindexter** [hashicorp/terraform-provider-aws#16533]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform CLI and Terraform AWS Provider Version
Terraform v0.13.5
+ provider registry.terraform.io/hashicorp/aws v3.19.0

### Affected Resource(s)

* aws_imagebuilder_image_recipe

### Terraform Configuration Files

```hcl
resource "aws_imagebuilder_image_recipe" "test" {
  name         = "test"
  parent_image = "arn:${data.aws_partition.current.partition}:imagebuilder:${data.aws_region.current.name}:aws:image/amazon-linux-2-ecs-optimized-x86/x.x.x"
  version      = "1.0.0"

  component {
    component_arn = "arn:aws:imagebuilder:us-west-2:aws:component/aws-cli-version-2-linux/1.0.0"
  }

  component {
    component_arn = "arn:aws:imagebuilder:us-west-2:aws:component/update-linux/1.0.0"
  }
}
```

### Expected Behavior

The components are added to the recipe in the order specified in the resource.

### Actual Behavior

The components are added to the recipe in a random order.

### Steps to Reproduce

1. `terraform apply`

### Important Factoids

I think this is because the components are TypeSet, which doesn't preserve order. But the order is significant: if a component depends on another component having run, it will fail if the order is random.

### References

https://docs.aws.amazon.com/imagebuilder/latest/userguide/how-image-builder-works.html#image-builder-components

* #0000

**Comment by rlove** [comment]
Yesterday I was bitten by two side effects.
1. Component tests don't always show up as tests unless they are all at bottom of the list.
2. Dependencies on other components. (as already mentioned)

It should be noted you can't change order after creation.   So there is no way to fix this post terraform run.

**Comment by bflad** [comment]
Fix submitted: https://github.com/hashicorp/terraform-provider-aws/pull/16566

**Comment by bflad** [comment]
The fix for this has been merged and will release with version 3.21.0 of the Terraform AWS Provider, likely tomorrow. 👍
